internal/lagoon/lagoonversion.go

```diff
@@ -0,0 +1,47 @@
+// Package lagoon implements high-level functions for interacting with the
+// Lagoon API.
+package lagoon
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/amazeeio/lagoon-cli/internal/schema"
+)
+
+// APIVersion interface contains methods for getting info on the current user of lagoon.
+type APIVersion interface {
+	LagoonAPIVersion(ctx context.Context, apiVersion *map[string]string) error
+	LagoonSchema(ctx context.Context, lagoonSchema *schema.LagoonSchema) error
+}
+
+// GetLagoonAPIVersion gets info on the current API version of lagoonm, supported in lagoon v1.4.1+
+func GetLagoonAPIVersion(ctx context.Context,
```

Comment:
Rather than doing the `json.Marshal` / `json.Unmarshal`, you could let the `graphql` package do this instead. Then this function can just be:
```
func GetLagoonAPIVersion(ctx context.Context, l APIVersion) (*schema.LagoonVersion, error) {
       apiVersion := schema.LagoonVersion{}
       return &apiVersion, l.LagoonAPIVersion(ctx, &apiVersion)
 }
```
Needs a tweak to the interface too:
```
LagoonAPIVersion(ctx context.Context, apiVersion *schema.LagoonVersion) error
```

Reply:
TIL